lxd/storage/drivers/driver_btrfs.go

```diff
@@ -192,7 +192,7 @@ func (d *btrfs) Create() error {
 		}
 	} else if d.config["source"] != "" {
 		hostPath := shared.HostPath(d.config["source"])
-		if d.isSubvolume(hostPath) {
+		if btrfsIsSubVolume(hostPath) {
```

Comment:
I would prefer we keep the d.isSubvolume function and (separately) remove the `btrfsIsSubVolume`.
This keeps the btrfs functionality inside the btrfs driver.

The fact that `btrfsIsSubVolume` is being called, indirectly, from `func (d *lxc) handleIdmappedStorage()` is a code smell that we need to get rid of really.

If you see patterns like:

``go
if storageType == "zfs" {
			err = nextIdmap.ShiftRootfs(d.RootfsPath(), storageDrivers.ShiftZFSSkipper)
		} else if storageType == "btrfs" {
			err = storageDrivers.ShiftBtrfsRootfs(d.RootfsPath(), nextIdmap)
		} else {
			err = nextIdmap.ShiftRootfs(d.RootfsPath(), nil)
		}
```

In places other than the storage subsystem then it means we've not done a good enough job of moving the legacy scattered storage handling into the per-storage drivers.

Reply:
Okay, thanks for the hints. I was wondering why it was duplicated, but this explains it. 

`d.isSubVolume()` is now back in and consistently used by the Btrfs driver files. Additionally I added the WalkDir fixes also for `BTRFSSubVolumesGet` but left it as it is. The deprecation marking is there.